docs/embedded/tutorials/migrate-abs-to-spe.md

```diff
@@ -0,0 +1,320 @@
+        if (subFolder == null)
+        {
+            subFolder = await _gcm.CreateFolder(folderName, newFolderId);
+            ...
+        }
+        newFolderId = subFolder.Id;
+
+        relativePath = newPath;
+    }
+
+    return newFolderId;
+```
+
+### Check if item exists
+```C#
+    var item = await _graphClient.Drives[_containerId].Items[parentFolderId].ItemWithPath(itemPath).GetAsync();
+```
+
+### Create folder
+```C#
+    var folder = new DriveItem
+    {
+        Name = folderName,
+        Folder = new Folder(),
+        AdditionalData = new Dictionary<string, object>()
+        {
+            { "@microsoft.graph.conflictBehavior", "fail" }
+        }
+    };
+    var createdFolder = await _graphClient.Drives[_containerId].Items[parentFolderId].Children.PostAsync(folder);
+```
+
+### MigrateFile
+```C#
+    internal async void MigrateFile(Object stateInfo)
+    {
+        var fileStructure = (FileStructure)stateInfo;
+
+        // Check if file exists in destination
+        ...
+
+        // Migrate the file
+        bool result = await TransferBlobToSharePointAsync(fileStructure.blobName, fileStructure.parentFolderId);
+
+        // Signal the countdown event that a file has been migrated
+        _countdown.Signal();
+
+        return;
+    }
+```
+
+### TransferBlobToSharePointAsync
+```C#
+    Stream blobStream = await _abm.DownloadBlobStreamAsync(blobName);
+
+    string fileName = Path.GetFileName(blobName);
+    await _gcm.UploadStreamToSharePointAsync(blobStream, parentFolderId, fileName);
+```
+
+### DownloadBlobStreamAsync
+```C#
+    BlobClient blobClient = _containerClient.GetBlobClient(blobName);
+
+    MemoryStream memoryStream = new MemoryStream();
+    await blobClient.DownloadToAsync(memoryStream);
+    memoryStream.Position = 0; // Reset the stream position to the beginning
+```
+
+### UploadStreamToSharePointAsync
+```C#
+    int _maxChunkSize = 320 * 1024;
+
+    var uploadSessionRequestBody = new CreateUploadSessionPostRequestBody()
+    {
+        AdditionalData = new Dictionary<string, object>
+        {
+            // Fail is set here, so it doesn't get upload again if it already exist
+            { "@microsoft.graph.conflictBehavior", "fail" }
+        }
+    };
+
+    var uploadSession = await _graphClient.Drives[_containerId]
+        .Items[parentFolderId]
+        .ItemWithPath(fileName)
+        .CreateUploadSession
+        .PostAsync(uploadSessionRequestBody);
+
+    // The stream is the same stream from the downloading the blob
+    var fileUploadTask = new LargeFileUploadTask<DriveItem>(uploadSession, stream, _maxChunkSize, _graphClient.RequestAdapter);
+        IProgress<long> progress = new Progress<long>(prog => Console.WriteLine($"Uploaded {fileName} {prog} bytes"));
+
+    // Check uploadResult.UploadSucceeded to see if it is successful
+    var uploadResult = await fileUploadTask.UploadAsync(progress);
+```
```

Comment:
I think a final structure of all the stubs you included until now would be helpful, to see how everything plays together